From 6402dc41098e3f8482f76d961f63b33cc28a6063 Mon Sep 17 00:00:00 2001
From: Marek Szyprowski <m.szyprowski@samsung.com>
Date: Thu, 8 Aug 2013 11:24:33 +0200
Subject: [PATCH 0564/1302] drm/exynos: fimd: disable windows before
 registering driver

Bootloader might enable some windows, so ensure that all windows are
disabled before registering to the system. This solves IOMMU fault during
registration if bootloader enabled some FIMD windows.

Signed-off-by: Marek Szyprowski <m.szyprowski@samsung.com>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 drivers/gpu/drm/exynos/exynos_drm_fimd.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/exynos/exynos_drm_fimd.c b/drivers/gpu/drm/exynos/exynos_drm_fimd.c
index 8aec342..ec060cb 100644
--- a/drivers/gpu/drm/exynos/exynos_drm_fimd.c
+++ b/drivers/gpu/drm/exynos/exynos_drm_fimd.c
@@ -1021,8 +1021,11 @@ static int fimd_probe(struct platform_device *pdev)
 	DRM_DEBUG_KMS("pixel clock = %d, clkdiv = %d\n",
 			panel->timing.pixclock, ctx->clkdiv);
 
-	for (win = 0; win < WINDOWS_NR; win++)
+	for (win = 0; win < WINDOWS_NR; win++) {
 		fimd_clear_win(ctx, win);
+		fimd_win_disable(dev, win);
+	}
+	fimd_wait_for_vblank(dev);
 
 	exynos_drm_subdrv_register(subdrv);
 
-- 
1.8.3.2

